tox.ini

```diff
@@ -1,12 +1,17 @@
 [tox]
 isolated_build = true
-envlist = clean,py37-django22,py37-django32,{38,39,310}-django{22,32,40},report
+envlist =
+    clean
+    py37-django{22,32}
+    py{38,39,310}-django{22,32,40,41}
```

Comment:
the previous config was missing `py` for these versions - it would just generate e.g. `38-django22`, which I think doesn't affect the Python version. So my guess is your tests have not been running with different Python versions for these environments.